src/workerd/io/actor-cache.c++

```diff
@@ -25,6 +25,21 @@ static constexpr size_t MAX_ACTOR_STORAGE_RPC_WORDS = (16u << 20) / sizeof(capnp
 
 ActorCache::Hooks ActorCache::Hooks::DEFAULT;
 
+namespace {
+
+// Utility functions for recording latency metrics via a one-liner in the callers below.
+auto recordStorageRead(ActorCache::Hooks& hooks, const kj::MonotonicClock& clock) {
+  auto start = clock.now();
```

Comment:
Sorry for asking late, but I'm a bit confused by how this works. I figure we start running immediately since we invoke when `attach()`ing, but then wouldn't the `kj::defer` run almost immediately after?

Reply:
Exactly!  We return the `kj::defer()`, so it becomes the responsibility of the caller to destroy it.